src/main/java/com/epam/parso/impl/SasFileParser.java

```diff
@@ -337,16 +338,14 @@ private void processSasFileHeader(String builderEncoding) throws IOException {
         }
 
         if (sasFileStream != null) {
-            try {
-                int bytesLeft = sasFileProperties.getHeaderLength() - currentFilePosition;
-                long actuallySkipped = 0;
-                while (actuallySkipped < bytesLeft) {
-                    actuallySkipped += sasFileStream.skip(bytesLeft - actuallySkipped);
-                }
-                currentFilePosition = 0;
-            } catch (IOException e) {
-                throw new IOException(EMPTY_INPUT_STREAM);
+            int bytesLeft = sasFileProperties.getHeaderLength() - currentFilePosition;
+            int skipped = sasFileStream.skipBytes(bytesLeft);
```

Comment:
I think this is a bit risky -- in some rare cases this would skip less bytes than it was asked to, because of network latency for example, so ideally this should be something like skipFully in IOUtils. As we don't really want to bring another dependency, probably it makes sense to copy IOUtils' logic (without some IFs, etc, of course). Thoughts?

Reply:
Sorry, just seeing this now.  Yes, I completely agree.

Reply:
It looks like DataInputStream's skipBytes does do the while...loop so that's not a problem...I don't think.

What is a problem is that FileInputStream can lie about bytes skipped, so y, I'll copy and paste from IOUtils.skipFully().

Reply:
From the decompiler in Java 8 on DataInputStream
```    public final int skipBytes(int n) throws IOException {
        int total = 0;
        int cur = 0;

        while ((total<n) && ((cur = (int) in.skip(n-total)) > 0)) {
            total += cur;
        }

        return total;
    }
```

Reply:
What I don't like about the current version is that if `readFully` fails, then eof=true and the user will get a null on next row instead of an EOFException, but if skip fails, then there's an exception.

Any preference/recommendation?

Reply:
Yes, agree, so I see the reason in your suggested change, so I would avoid readFully in favor of IOUtils.skipFully (it's just better than streams' skip/read methods) so perhaps just copy and paste those IOUtils methods as the licence is the same?